collections/duties.js

```diff
@@ -57,5 +57,35 @@ function getDutie(name, id) {
   });
 }
 
+function deleteDutie(id) {
+  const resHandler = require("../start").server.getResHandler();
+  const collection = require("../start").dutiesCollection;
+
+  if (!utils.isMatchesRegex(/[0-9a-fA-F]{24}/, id)) {
+    resHandler.routeNotFound("the id that was sent is not legal, it needs to be of 24 chars in alpha-numeric");
+    return;
+  }
+
+  let searchedParams = {
+    "_id": objectId(id)
+  };
+
+  collection.find(searchedParams).toArray(function (err, doc) {
+    let isDutieExists = (doc.length == 1);
+    let isDutieDeletable = isDutieExists && (doc[0]["soldiers"].length == 0);
+
+    if (isDutieDeletable) {
+      collection.deleteOne(searchedParams, function (err, result) {
+        resHandler.success();
+      });
+    } else if (isDutieExists && !isDutieDeletable) {
+      resHandler.notValidData("dutie cannot be deleted beacuse its already assigned");
+    } else {
+      resHandler.routeNotFound("the referenced dutie doesnt exist");
```

Comment:
this is tricky now. but if a duty doesn't exist then deleting it alwauys succeeds. why? because an error tells the user "something F'd up, try again", but what did he try to do here? make sure that the duty is gone. but it doesn't exist to begin with....so success! giving an error message here would confuse the user. just say "oh that? yeah yeah. totally gone good job"